### DISCUSSION TITLE
Is it possible to create a custom reusable HTML block?

### DISCUSSION THREAD

**Discussion by vlad-ivanov-name** [squidfunk/mkdocs-material#7451] [Q&A] [answered]
I might be missing something obvious, but I'd like to be able to create a new reusable component that I would then be able to instantiate several times with different parameters. The parameters wouldn't need any complex processing, so it's as simple as something like:

```
<mymagicblock label="test"></mymagicblock>
```

Triggering html instantiation that is defined somewhere in `custom_dir`.

What would be the simplest way to achieve that?

**Comment by alexvoss** [comment, marked as answer]
I had a look and it seems that Python Markdown passes not only HTML elements through but also custom elements like the one you would like to use. If you can then animate it in the way you want using some [added JavaScript](https://squidfunk.github.io/mkdocs-material/customization/?h=extra+javascript#additional-javascript) then that should work fine. 

If Javascript does not work for you or your use case, you could try [the Macros plugin](https://mkdocs-macros-plugin.readthedocs.io/en/latest/), which allows you to replace a placeholder with content using Python.

**Reply by squidfunk** [reply]
Interestingly, reusable components is something that is on our (yet unshared) roadmap 🤫
